src/ipa/rpi/controller/rpi/awb.h

```diff
@@ -1,42 +1,46 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright (C) 2019, Raspberry Pi Ltd
+ * Copyright (C) 2025, Raspberry Pi Ltd
  *
  * AWB control algorithm
  */
 #pragma once
 
-#include <mutex>
 #include <condition_variable>
+#include <mutex>
 #include <thread>
 
-#include <libcamera/geometry.h>
-
 #include "../awb_algorithm.h"
 #include "../awb_status.h"
-#include "../statistics.h"
-
 #include "libipa/pwl.h"
 
 namespace RPiController {
 
-/* Control algorithm to perform AWB calculations. */
-
 struct AwbMode {
-	int read(const libcamera::YamlObject &params);
+	int read(const libcamera::YamlObject &params)
+	{
+		auto value = params["lo"].get<double>();
+		if (!value)
+			return -EINVAL;
+		ctLo = *value;
+
+		value = params["hi"].get<double>();
+		if (!value)
+			return -EINVAL;
+		ctHi = *value;
+
+		return 0;
+	}
 	double ctLo; /* low CT value for search */
 	double ctHi; /* high CT value for search */
 };
 
-struct AwbPrior {
-	int read(const libcamera::YamlObject &params);
-	double lux; /* lux level */
-	libcamera::ipa::Pwl prior; /* maps CT to prior log likelihood for this lux level */
-};
-
 struct AwbConfig {
-	AwbConfig() : defaultMode(nullptr) {}
+	AwbConfig()
+		: defaultMode(nullptr) {}
 	int read(const libcamera::YamlObject &params);
+	bool hasCtCurve() const;
```

Comment:
It feels a teeny bit weird to have both a data member and a member function called `hasCtCurve`! Perhaps you could set the data member just once, depending on whether the curve is empty, and then isn't the function version superfluous?